src/api/realtime/realTimeModule.ts

```diff
@@ -250,15 +253,35 @@ export class RealTimeModule implements IModule {
 
       RTCResources.forEach((resource) => resource.prototype.webSocket = this.websocket);
 
+      // listen to onclose event and check it's needed to reconnect
+      this.websocket.onclose = (error) => this.reconnect(error);
+
       return new Promise((resolve, reject) => {
         this.websocket.onopen = resolve;
         this.websocket.onerror = () => reject(new Error(MESSAGE.RTC.CONNECTION_FAILED));
       });
     })
-      .then(() => websocket.start(this.websocket, () => this.tokenManager.token().toPromise()))
-      .then(() => this);
+      .then(() => websocket.start(this.websocket, () => this.tokenManager.token().toPromise(), reconnection));
   }
 
+  /**
+   * Reconnect the websocket when it got unexpectedly closed
+   *
+   * @internal
+   */
+  private reconnect(event: CloseEvent) {
+    if (event instanceof CloseEvent && event.code !== 4999) {
```

Comment:
magic number: `4999`

Reply:
~and according to https://developer.mozilla.org/en-US/docs/Web/API/CloseEvent the full range of 4000-4999 is for custom events...~ nevermind I later saw _you_ send the `4999`... so I guess using some constant for this magic number indeed can help :smile:

Reply:
😊  yes will make a constant for that, should be better.